### PR TITLE
Clean up install config files and ensure find_dependency is called as appropriate

### LINKED ISSUE #449: Missing find_dependency call in fclConfig.cmake when imported targets are used in target_link_libraries
In the CMake build scripts of `fcl`, if the imported targets  `Eigen3::Eigen`, `ccd` or `octomap` are defined, they are used in `target_link_libraries` with `PUBLIC` visibility (see https://github.com/flexible-collision-library/fcl/pull/196). When this  happens, then in the `fclConfig.cmake` script the appropriate [`find_dependency`](https://cmake.org/cmake/help/v3.12/module/CMakeFindDependencyMacro.html) script should be added, to ensure that the imported  targets are properly defined also in the downstream project that consume `fcl` via `find_package(fcl)`.

### PR DESCRIPTION
**Pull request by jamiesnape**
Closes #449. FYI @traversaro.

---
This change is [Reviewable](https://reviewable.io/reviews/flexible-collision-library/fcl/452)

### CODE REVIEW

File: fcl-config.cmake.in [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Generated by CMake @CMAKE_VERSION@ for @PROJECT_NAME@.
+
+@PACKAGE_INIT@
+
+set(FCL_VERSION "@FCL_VERSION@")
+set(FCL_ABI_VERSION "@FCL_ABI_VERSION@")
+
+set_and_check(FCL_INCLUDE_DIRS "@PACKAGE_CMAKE_INSTALL_INCLUDEDIR@")
+set(FCL_LIBRARIES @PROJECT_NAME@)
+set_and_check(FCL_LIBRARY_DIRS "@PACKAGE_CMAKE_INSTALL_LIBDIR@")
```
Comment (traversaro) [inline fcl-config.cmake.in]: There is any reason to define the `FCL_INCLUDE_DIRS` and `FCL_LIBRARY_DIRS` variables? If necessary, these can be  easily be obtained via the `fcl` imported target.

Comment (jamiesnape) [reply fcl-config.cmake.in]: Partly tradition, but also makes for a clearer error message when the path is accidentally cleaned IMO.

**Comment by jamiesnape** [comment]
You should probably cherry-pick #450 as well?

**Comment by jamiesnape** [comment]
Yes, though we need to create a release branch (`fcl-6.0`) from a13c681e41eb8180cba7d4fd32637511f588cb82, cherry pick #450, merge this into the release branch, tag that `0.6.1`, and merge the release branch into` master`.

**Comment by jamiesnape** [comment]
So the first response to that would be why was the previous release tagged `0.6.0`. What do `x = 0`, `y = 6`, and `z = 0` mean?

**Comment by jamiesnape** [comment]
(don't squash the two commits)

**Pull request merged**